examples/tpm2-seal-unseal/main.go

```diff
@@ -0,0 +1,281 @@
+	// https://trustedcomputinggroup.org/wp-content/uploads/Credential_Profile_EK_V2.0_R14_published.pdf
+	// Shared SRK template based off of EK template and specified in:
+	// https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-v2.0-Provisioning-Guidance-Published-v1r1.pdf
+	srkTemplate = tpm2.Public{
+		Type:       tpm2.AlgRSA,
+		NameAlg:    tpm2.AlgSHA256,
+		Attributes: tpm2.FlagFixedTPM | tpm2.FlagFixedParent | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth | tpm2.FlagRestricted | tpm2.FlagDecrypt | tpm2.FlagNoDA,
+		AuthPolicy: nil,
+		RSAParameters: &tpm2.RSAParams{
+			Symmetric: &tpm2.SymScheme{
+				Alg:     tpm2.AlgAES,
+				KeyBits: 128,
+				Mode:    tpm2.AlgCFB,
+			},
+			KeyBits:    2048,
+			Exponent:   0,
+			ModulusRaw: make([]byte, 256),
+		},
+	}
+)
+
+func main() {
+	tpmPathFlag := flag.String("tpm-path", "/dev/tpm0", "Path to the TPM device (character device or a Unix socket).")
+	pcrFlag := flag.Int("pcr", -1, "PCR to seal data to. Must be within [0, 23].")
+	flag.Parse()
+
+	if *pcrFlag < 0 || *pcrFlag > 23 {
+		fmt.Fprintf(os.Stderr, "Invalid flag 'pcr': value %d is out of range", *pcrFlag)
+		os.Exit(1)
+	}
+
+	errs := run(*pcrFlag, *tpmPathFlag)
+	for _, err := range errs {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+	}
+	if len(errs) > 0 {
+		os.Exit(1)
+	}
+}
+
+func run(pcr int, tpmPath string) (errs []error) {
+	// Open the TPM
+	rwc, err := tpm2.OpenTPM(tpmPath)
+	if err != nil {
+		return append(errs, fmt.Errorf("can't open TPM %q: %v", tpmPath, err))
+	}
+	defer func() {
+		if err := rwc.Close(); err != nil {
+			errs = append(errs, fmt.Errorf("can't close TPM %q: %v", tpmPath, err))
+		}
+	}()
+
+	// Create the parent key against which to seal the data
+	srkPassword := ""
+	srkHandle, _, err := tpm2.CreatePrimary(rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", srkPassword, srkTemplate)
+	if err != nil {
+		return append(errs, fmt.Errorf("can't create primary key: %v", err))
+	}
+	defer func() {
+		if err := tpm2.FlushContext(rwc, srkHandle); err != nil {
+			errs = append(errs, fmt.Errorf("unable to flush SRK handle %q: %v", srkHandle, err))
+		}
+	}()
+	fmt.Printf("Created parent key with handle: 0x%x\n", srkHandle)
+
+	// Note the value of the pcr against which we will seal the data
+	pcrVal, err := readPCR(rwc, pcr)
+	if err != nil {
+		return append(errs, fmt.Errorf("unable to read PCR: %v", err))
+	}
+	fmt.Printf("PCR %v value: 0x%x\n", pcr, pcrVal)
+
+	// Get the authorization policy that will protect the data to be sealed
+	objectPassword := "objectPassword"
+	sessHandle, policy, err := policyPCRPasswordSession(rwc, pcr, objectPassword)
+	if sessHandle != tpm2.HandleNull {
+		if flushErr := tpm2.FlushContext(rwc, sessHandle); flushErr != nil {
+			errs = append(errs, fmt.Errorf("unable to flush session: %v", flushErr))
+		} else {
+			sessHandle = tpm2.HandleNull
+		}
+	}
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to get policy: %v", err))
+	}
+	if len(errs) > 0 {
```

Comment:
return right away in case of errors above instead

Reply:
Do you mean 

```errs = append(errs, fmt.Errorf("unable to get policy: %v", err))```

should be

```return append(errs, fmt.Errorf("unable to get policy: %v", err))```

?

I think it is cleaner to encapsulate the idea "if the previous command failed in any way then abort the current function" to a single place...WDYT?

Reply:
Yeah, that's what I meant.

If the previous command failed in any way, it should have aborted. No need to complicate the code to have fewer returns

Reply:
i see it the other way, but I'll change this